### PR TITLE
build and use the controller testing

### PR DESCRIPTION
**Pull request by k-rister**

### CODE REVIEW

File: .github/actions/integration-tests/action.yml [hunk 1]
```diff
@@ -33,6 +33,10 @@ inputs:
     description: "Which rickshaw endpoint to exercise"
     required: false
     default: "remotehost"
+  ci_build_controller:
+    description: "Should the crucible controller be rebuilt and tested"
+    required: false
+    default: "no"
```
Comment (rafaelfolco) [inline .github/actions/integration-tests/action.yml:39]: here default=no

File: .github/actions/integration-tests/base [hunk 2]
```diff
@@ -31,6 +31,19 @@ function log_rc {
...
 }
 
+function validate_ci_build_controller {
+    case "${CI_BUILD_CONTROLLER}" in
+        yes|no)
+            echo "CI Build Controller is '${CI_BUILD_CONTROLLER}'"
+            echo
+            ;;
+        *)
+            echo "ERROR: Unknown CI_BUILD_CONTROLLER value [${CI_BUILD_CONTROLLER}]"
```
Comment (rafaelfolco) [inline .github/actions/integration-tests/base:41]: you could just print a warning and default to no

**Review by rafaelfolco — APPROVED** [review]

File: .github/actions/integration-tests/base (hunk 2 above)
Comment (k-rister) [reply .github/actions/integration-tests/base:41]: This is just being consistent with some of the other validators that are already present.  They don't make a lot of sense when running the CI via Github runners (self hosted or not) but it is possible to execute the CI manually outside of the Github environment and so parameter validation is important for that use case.

File: .github/actions/integration-tests/action.yml (hunk 1 above)
Comment (k-rister) [reply .github/actions/integration-tests/action.yml:39]: I'm not sure I follow what you mean here, can you elaborate?

**Pull request merged**